libfprint-tod: 1.90.7+git20210222+tod1 -> 1.94.9+tod1

danieldk opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

This also fixes the build issue introduced by #388631.

Checked to work on a ThinkPad T14 Gen 5 AMD with a goodix fingerprint scanner.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

danieldk committed:
libfprint-tod: 1.90.7+git20210222+tod1 -> 1.94.9+tod1

This also fixes the build issue introduced by #388631.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5275

VuiMuich commented:
This breaks build for `libfprint-2-tod1-vfs0090` to keep readability I don't post the logs here, as I alreade added them in #390101 in the "Additional Context" section.

danieldk committed:
libfprint-2-tod1-vfs0090: mark broken

This package does not compile against recent libfprint-tod anymore and
upstream hasn't seen updates since 2020.

danieldk commented:
> This breaks build for `libfprint-2-tod1-vfs0090` to keep readability I don't post the logs here, as I alreade added them in #390101 in the "Additional Context" section.

Thanks! I have updated the pr to mark it broken `libfprint-2-tod1-vfs0090`. Upstream hasn't updated since 2020, so unless someone updates it, it seems left behind by the fprint ecosystem.

pineapplehunter commented:
Here is an overlay for users of `libfprint-2-tod1-vfs0090` once this PR takes effect.
<details>
<summary>overlay</summary>

```nix
final: prev: {
  libfprint-2-tod1-vfs0090 =
    (prev.libfprint-2-tod1-vfs0090.override {
      libfprint-tod = prev.libfprint-tod.overrideAttrs (old: rec {
        version = "1.90.7+git20210222+tod1";
        src = old.src.overrideAttrs {
          rev = "v${version}";
          outputHash = "0cj7iy5799pchyzqqncpkhibkq012g3bdpn18pfb19nm43svhn4j";
          outputHashAlgo = "sha256";
        };
        buildInputs = (old.buildInputs or [ ]) ++ [ final.nss ];
        mesonFlags = [
          "-Ddrivers=all"
          "-Dudev_hwdb_dir=${placeholder "out"}/lib/udev/hwdb.d"
        ];
      });
    }).overrideAttrs
      { meta.broken = false; };
};
```
</details>

VuiMuich commented:
I had a brief look and it seems the upstream maintainer is generally active, so I'll look into opening an issue upstream.

danieldk commented:
> I had a brief look and it seems the upstream maintainer is generally active, so I'll look into opening an issue upstream.

Good idea, it's certainly worth asking!

VuiMuich commented:
:face_with_diagonal_mouth: looks like freedesktops gitlab instance is N/A due to migration.. will have a look at it asap.

MJWcodr commented:
Is there anything new here or help needed?

kirillrdy reviewed (approved):
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 388905`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>libfprint-2-tod1-vfs0090</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>fprintd-tod</li>
    <li>fprintd-tod.devdoc</li>
    <li>libfprint-2-tod1-broadcom</li>
    <li>libfprint-2-tod1-elan</li>
    <li>libfprint-2-tod1-goodix</li>
    <li>libfprint-2-tod1-goodix-550a</li>
    <li>libfprint-tod</li>
    <li>libfprint-tod.devdoc</li>
  </ul>
</details>

kirillrdy commented:
@danieldk  @pineapplehunter  is there is a fix for libfprint-2-tod1-vfs0090 feel free to ping me
